File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        List<Integer> answers = new ArrayList<>();
        Queue<Integer> days = new LinkedList<>();
        
        //각 작업 당 며칠이 걸리는지 계산 후 큐 초기화 (큐 = days)
        for(int i=0; i<progresses.length; i++){
            int day = (100-progresses[i])/speeds[i];
            if((100-progresses[i])%speeds[i]!=0) day++;
            days.offer(day);
        }
        
        while(!days.isEmpty()){
            int cnt = 1;
            int fir = days.poll(); 
            //앞에 있는 작업이 완료되었을 때, 그 후에 몇 개의 작업이 같이 배포되는지 계산
            while(!days.isEmpty()){
                if(days.peek()<=fir){
                    cnt++; days.poll();
                }else break;
            }
            answers.add(cnt);
        }
        
        int[] answer = new int[answers.size()];
        for(int i=0; i<answers.size(); i++) 
            answer[i] = answers.get(i);
        
        return answer;
    }
}

File: 프로그래머스/2/42586. 기능개발/기능개발.cpp
#include <string>
#include <vector>
#include <queue>
#include <iostream>

using namespace std;

struct Prog{
    int progress;
    int speed;
};
vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    queue<Prog> q;
    for(int i=0; i<speeds.size(); i++) q.push({progresses[i],speeds[i]});
    
    int day = 0;
    while(!q.empty()){
        Prog prog = q.front();
        q.pop();
        if((100-prog.progress)%prog.speed==0) 
            day = (100-prog.progress)/prog.speed;
        else day = (100-prog.progress)/prog.speed+1;
        int cnt = 1;
        while(true){
            Prog cur = q.front();
            //cout<<"day:"<<day<<endl;
            //cout<<"sum:"<<cur.progress+day*cur.speed<<endl;
            if(cur.progress+day*cur.speed>=100){
                cnt++;
                q.pop();
            } 
            else break;
        }
        answer.push_back(cnt);
    }
    
    return answer;
}